### PR TITLE
Decode and use ESC_TELEMETRY_X_TO_Y mavlink packets as an AP_ESC_Telem_Backend

### PR DESCRIPTION
**Pull request by amilcarlucas**
An alternative to #19666, using a completely different approach.

```
Binary Name      Text [B]        Data [B]     BSS (B)      Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  --------------  -----------  -----------  ----------------------------  -------------------------
arducopter-heli  672 (+0.0388%)  0 (0.0000%)  0 (0.0000%)  672 (+0.0387%)                                   228792
blimp            672 (+0.0527%)  0 (0.0000%)  0 (0.0000%)  672 (+0.0527%)                                   689084
antennatracker   672 (+0.0500%)  0 (0.0000%)  0 (0.0000%)  672 (+0.0499%)                                   619868
ardusub          672 (+0.0431%)  0 (0.0000%)  0 (0.0000%)  672 (+0.0431%)                                   404548
arducopter       672 (+0.0383%)  0 (0.0000%)  0 (0.0000%)  672 (+0.0382%)                                   206400
arduplane        672 (+0.0382%)  0 (0.0000%)  0 (0.0000%)  672 (+0.0382%)                                   204612
ardurover        672 (+0.0425%)  0 (0.0000%)  0 (0.0000%)  672 (+0.0424%)                                   380668
```

### CODE REVIEW

**Review by andyp1per — COMMENTED** [review]
Very cool. How do you switch it off?

**Comment by amilcarlucas** [comment]
If no ESC_TELEMETRY_X_TO_Y mavlink packets come in, it will stop updating the AP_ESC_Telem.

Does that answer your question? Or Are you asking to use a preprocessor compile macro like "HAL_MAVLINK_ESC_TELEM_INPUT_ENABLED" ?

**Comment by tridge** [comment]
> If no ESC_TELEMETRY_X_TO_Y mavlink packets come in, it will stop updating the AP_ESC_Telem.

seems a bit dangerous for something like two vehicles doing followme, or a swarm. They will each bounce packets back and forth between them

File: libraries/GCS_MAVLink/MAVLink_routing.cpp [hunk 1]
```diff
@@ -122,6 +123,105 @@ bool MAVLink_routing::check_and_forward(mavlink_channel_t in_channel, const mavl
         return true;
     }
 
+#if HAL_WITH_ESC_TELEM
```
Comment (tridge) [inline libraries/GCS_MAVLink/MAVLink_routing.cpp:126]: this is really in the wrong place, should be in AP_ESC_Telem_Backend, and only when enabled

**Review by tridge — CHANGES_REQUESTED** [review]
we could end up bouncing packets back and forth infiniitely

File: libraries/GCS_MAVLink/MAVLink_routing.cpp (hunk 1 above)
Comment (GithubKhang) [reply libraries/GCS_MAVLink/MAVLink_routing.cpp:126]: Hi @tridge, Could you explain to me why these lines are in the wrong place?  I have built this code and It works fine. I'm sorry. I'm a newbie for this.

**Comment by amilcarlucas** [comment]
How exaclly can I prevent the packets from bouncing?

File: libraries/GCS_MAVLink/MAVLink_routing.cpp (hunk 1 above)
Comment (tridge) [reply libraries/GCS_MAVLink/MAVLink_routing.cpp:139]: these messages are not targeted to a specific sysid, so I don't think they are safe to use in this way.
imagine 2 vehicles on the radio link, both generating these msgs, they would consume each others esc telem msgs and get the wrong RPMs to notch filter

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by amilcarlucas** [comment]
Thanks for the clarifications, closing it now.

**Pull request closed without merging**